File: tests/sdk/nodejs/test/controltower-stub.test.mjs
import { test } from "node:test";
import assert from "node:assert/strict";
import { requireReady, signedJsonTarget } from "../lib/helpers.mjs";

test("Control Tower ListLandingZones empty and GetLandingZone not found", async (t) => {
  if (!(await requireReady(t))) return;

  const listed = await signedJsonTarget(
    "controltower",
    "ControlTower.ListLandingZones",
    {},
  );
  assert.equal(
    listed.status,
    200,
    `ListLandingZones status=${listed.status} body=${listed.body}`,
  );
  const zones = listed.json?.landingZones || [];
  assert.equal(zones.length, 0, `landingZones=${JSON.stringify(zones)}`);

  const got = await signedJsonTarget(
    "controltower",
    "ControlTower.GetLandingZone",
    {
      landingZoneIdentifier:
        "arn:aws:controltower:us-east-1:000000000001:landingzone/lz-deadbeef",
    },
  );
  assert.equal(
    got.status,
    400,
    `GetLandingZone status=${got.status} body=${got.body} want 400`,
  );
  assert.match(
    got.body,
    /ResourceNotFoundException/,
    `GetLandingZone body=${got.body}`,
  );
});

File: tests/sdk/go/controltower_stub_test.go
package sdk_test

import (
	"strings"
	"testing"
)

func TestControlTowerListEmptyGetNotFound(t *testing.T) {
	requireReady(t)

	listStatus, listBody, listParsed := signedJSONTarget(t, "controltower", "ControlTower.ListLandingZones", map[string]any{})
	if listStatus != 200 {
		t.Fatalf("ListLandingZones status=%d body=%s", listStatus, listBody)
	}
	zones, _ := listParsed["landingZones"].([]any)
	if len(zones) != 0 {
		t.Fatalf("landingZones=%v want empty", zones)
	}

	getStatus, getBody, _ := signedJSONTarget(t, "controltower", "ControlTower.GetLandingZone", map[string]any{
		"landingZoneIdentifier": "arn:aws:controltower:us-east-1:000000000001:landingzone/lz-deadbeef",
	})
	if getStatus != 400 {
		t.Fatalf("GetLandingZone status=%d body=%s want 400", getStatus, getBody)
	}
	if !strings.Contains(string(getBody), "ResourceNotFoundException") {
		t.Fatalf("GetLandingZone body=%s want ResourceNotFoundException", getBody)
	}
}
